exporter/collector/config.go

```diff
@@ -50,6 +50,16 @@ type MetricConfig struct {
 	// If a metric belongs to one of these domains it does not get a prefix.
 	KnownDomains []string `mapstructure:"known_domains"`
 
+	// If true, set the instrumentation_source and instrumentation_version
+	// labels. Defaults to true.
+	InstrumentationLibraryLabels bool `mapstructure:"instrumentation_library_labels"`
+
+	// Set of metric name prefixes to add instrumentation_source and
+	// instrumentation_version labels to. Defaults to custom metric domains:
+	// "custom.googleapis.com/", "external.googleapis.com/",
+	// "prometheus.googleapis.com/", "workload.googleapis.com/"
+	InstrumentationLibraryPrefixes []string `mapstructure:"instrumentation_library_domains"`
```

Comment:
Can this be hard-coded instead of configurable?  New custom domains are fairly rare.

Reply:
For similar reasons to why we are allowing configurable metric domains in general, I'd like to have this exposed as an escape hatch.   This should help us avoid getting into a scenario in the future e.g. w/ open-census where we had lots of hard coded logic that become out-of-date and hard to adjust.  Specifically, usually changes to something like this come with a narrow time window.

I'd actually *change* this configuration, however, to be "custom_metric_domains".   Specifically, there's more than IL that needs to diverge its behavior based on whether or not something is in a known custom metric domain.

Reply:
I think `custom_metric_domains` is an improvement, but I still prefer not adding configuration for which we don't have a use-case.  If the list becomes outdated, the solution is to update the list, regardless of whether it is configurable or not.  Unnecessary config just makes using the exporter more complex for users.

> Specifically, there's more than IL that needs to diverge its behavior based on whether or not something is in a known custom metric domain.

What other behavior?  This is the first I know of, since CreateServiceTimeSeries is explicit, rather than based on domain.

Reply:
discussed offline: we are going to remove this logic entirely and rely on the instrumentation_library_labels flag